File: app/src/main/java/waldrapps/plannit/adapters/ContactAdapter.java
package waldrapps.plannit.adapters;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Calendar;
import java.util.List;

import waldrapps.plannit.Contact;
import waldrapps.plannit.R;
import waldrapps.plannit.activities.HomeScreenActivity;

public class ContactAdapter extends RecyclerView.Adapter<ContactAdapter.ContactViewHolder> {

    public class ContactViewHolder extends RecyclerView.ViewHolder {
        public TextView name;
        public ImageView icon;

        ContactViewHolder(View view) {
            super(view);
            name = view.findViewById(R.id.contact_name);
            icon = view.findViewById(R.id.person_icon);
        }
    }

    private final LayoutInflater inflater;
    //Cached copy of contacts
    private List<Contact> contacts;

    public ContactAdapter(Context context) { inflater = LayoutInflater.from(context); }

    @NonNull
    @Override
    public ContactAdapter.ContactViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = inflater.inflate(R.layout.list_item_contact, parent, false);
        view.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(v.getContext(), "Recycler",
                        Toast.LENGTH_LONG).show();
            }
        });
        return new ContactViewHolder(view);
    }

    public void onBindViewHolder(@NonNull final ContactViewHolder holder, final int position) {
        if(contacts != null) {
            Contact current = contacts.get(position);
            //Find current date and time
            Calendar c = Calendar.getInstance();
            c.set(Calendar.DAY_OF_WEEK, Calendar.MONDAY);
            c.set(Calendar.HOUR_OF_DAY, 20);
            c.set(Calendar.MINUTE, 21);
            int day = c.get(Calendar.DAY_OF_WEEK);
            int hour = c.get(Calendar.HOUR_OF_DAY);
            int min = c.get(Calendar.MINUTE);
            int pos = -1;
            if((day >= Calendar.MONDAY) && (day <= Calendar.FRIDAY))
            {
                if(hour >= 8 && hour <= 21)
                {
                    pos = (28 * (day - 2)) + (2 * (hour - 8)) + (min / 30);
                }
            }

            //Set icon based on flag
            if (current.getFlag().equals("g")) {
                holder.icon.setImageResource(R.drawable.ic_people);
            }
            else {
                holder.icon.setImageResource(R.drawable.ic_person_white_24dp);
            }

            //Set name
            holder.name.setText(current.getName());

        } else {
            //Covers the case of data not being ready yet
            holder.name.setText(R.string.no_name);
        }
    }

    public void setContacts(List<Contact> contacts){
        this.contacts = contacts;
        notifyDataSetChanged();
    }

    public Contact getItem(int pos) {
        return contacts.get(pos);
    }

    @Override
    public int getItemCount() {
        if (contacts != null) {
            return contacts.size();
        }
        else {
            return 0;
        }
    }
}

File: app/src/main/java/waldrapps/plannit/adapters/ContactAdapter.kt
package waldrapps.plannit.adapters

import android.content.Context
import android.graphics.Color
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import kotlinx.android.synthetic.main.item_contact.view.*
import waldrapps.plannit.Contact
import waldrapps.plannit.R
import waldrapps.plannit.activities.GroupActivity
import waldrapps.plannit.fragments.GroupFragment
import waldrapps.plannit.utils.getLayoutInflater

class ContactAdapter(
        val context : Context,
        private val clickListener: (contact: Contact) -> Unit,
        private val checkBoxListener : (contact: Contact, b: Boolean) -> Unit
) : RecyclerView.Adapter<ContactAdapter.ViewHolder>() {

    private var contactsList: List<Contact>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = parent.getLayoutInflater().inflate(R.layout.item_contact, parent, false)
        if(context is GroupActivity) {
            view.checkbox.visibility = View.VISIBLE
        }
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(context, contactsList!![position], clickListener, checkBoxListener)
    }

    fun updateContacts(contacts: List<Contact>) {
        contactsList = contacts
        notifyDataSetChanged()
    }

    fun getItem(pos: Int): Contact {
        return contactsList!![pos]
    }

    override fun getItemCount(): Int {
        return if (contactsList != null) {
            contactsList!!.size
        } else {
            0
        }
    }

    class ViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
        fun bind(context: Context,
                contact: Contact,
                clickListener: (contact: Contact) -> Unit,
                checkBoxListener: (contact: Contact, b: Boolean) -> Unit?
        ) {
            view.card.setCardBackgroundColor(Color.parseColor(contact.color))
            view.contact_name.text = contact.name ?: "?"
            if(contact.flag == context.resources.getString(R.string.group_flag)) {
                view.person_icon.setImageResource(R.drawable.ic_people)
            } else {
                view.person_icon.setImageResource(R.drawable.ic_person_white_24dp)
            }
            view.setOnClickListener { clickListener(contact) }
            view.checkbox.setOnCheckedChangeListener { _: CompoundButton, b: Boolean -> checkBoxListener(contact, b)}
        }
    }
}